dash.el

```diff
@@ -2023,7 +2024,7 @@ SOURCE is a proper or improper list."
        ((cdr match-form)
         (cond
          ((and (symbolp (car match-form))
-               (functionp (dash--get-expand-function (car match-form))))
+               (dash--get-expand-function (car match-form)))
```

Comment:
I'm slightly undecided whether it's not better to call the `functionp` tests here (and other places this function is used), since the function is called `...get...` but here it serves as a predicate which might be a bit confusing.  And this matching code is quite complex, so maybe more verbosity is better.  WDYT?